### ISSUE TITLE
Also when trying to compile `varint.c` with `wcc` compiled with `gcc` this error message appear:

### ISSUE THREAD

**Issue by tyfkda** [tyfkda/xcc#87]
Also when trying to compile `varint.c` with `wcc` compiled with `gcc` this error message appear:
```
./wcc ../../xcc-dad/varint.c
*** stack smashing detected ***: <unknown> terminated
Aborted (core dumped)
```
Using valgrind to show where the code segfaults and changing the buffers size makes the error message disappear but the onine compiler still give the same error message:
```
git diff -u gen_wasm.c 
diff --git a/wasm/src/gen_wasm.c b/wasm/src/gen_wasm.c
index 9ed4d18..472b088 100644
--- a/wasm/src/gen_wasm.c
+++ b/wasm/src/gen_wasm.c
@@ -32,7 +32,7 @@ static void add_code(const unsigned char* buf, size_t size) {
 }
 
 void emit_leb128(DataStorage *data, ssize_t pos, int64_t val) {
-  unsigned char buf[5], *p = buf;
+  unsigned char buf[32], *p = buf;
   const int64_t MAX = 1 << 6;
   for (;;) {
     if (val < MAX && val >= -MAX) {
@@ -46,7 +46,7 @@ void emit_leb128(DataStorage *data, ssize_t pos, int64_t val) {
 }
 
 void emit_uleb128(DataStorage *data, ssize_t pos, uint64_t val) {
-  unsigned char buf[5], *p = buf;
+  unsigned char buf[32], *p = buf;
   const uint64_t MAX = 1 << 7;
   for (;;) {
     if (val < MAX) {
```

_Originally posted by @mingodad in https://github.com/tyfkda/xcc/issues/85#issuecomment-1230282415_

**Comment by tyfkda** [comment]
@mingodad Would you send a PR? Thanks!

**Comment by tyfkda** [comment]
Fixed in 981bea0
